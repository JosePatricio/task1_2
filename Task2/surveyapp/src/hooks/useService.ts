import AsyncStorage from '@react-native-community/async-storage';
import {useMemo} from 'react';
import {SurveyData, SurveyValue} from '../entities/interfaces';

export function useService() {
  const service = useMemo(
    () => ({
      saveSurvey: async (rateValue: SurveyValue) => {
        const storagedSurveys = await AsyncStorage.getItem('surveys');
        let surveys: SurveyData[] = [];

        const currentDate = new Date().getTime();
        if (!storagedSurveys) {
          surveys = [
            {id: surveys.length, survey: rateValue, date: currentDate},
          ];
        } else {
          surveys = JSON.parse(storagedSurveys);
          surveys.push({
            id: surveys.length,
            survey: rateValue,
            date: currentDate,
          });
        }

        await AsyncStorage.setItem('surveys', JSON.stringify(surveys));
      },
    }),

    [],
  );

  return {service};
}
