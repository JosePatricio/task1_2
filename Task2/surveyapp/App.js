/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { useState } from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
import { AirbnbRating } from 'react-native-ratings';
import { useService } from './src/hooks/useService';

const App = () => {
  const [rateValue, setRateValue] = useState<SurveyValue>({
    survey1: 0,
    survey2: 0,
  });
  const surveyRange = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10'];

  const {service} = useService();

  const onSubmitSurvey = async() => {
      await service.saveSurvey(rateValue);
      setRateValue({survey1: 0, survey2: 0});
      alert('Complete!'); 
  };

  const onClearSurvey = async() => {
    setRateValue({survey1: 0, survey2: 0});
  };

  return (
    
    <View style={styles.container}>
      <Text style={styles.title}>Mobile Survey App</Text>

      <View style={styles.reviewSection}>
        <Text style={styles.surveyText}>
          How was your sleep last night
        </Text>
         <AirbnbRating
          starContainerStyle={{flexWrap: 'wrap', padding: 5}}
          count={10}
          onFinishRating={(e)=>{
            setRateValue({...rateValue, survey1: e});
          }}
          reviews={surveyRange}
          defaultRating={rateValue.survey1}
          size={40}
          showRating={false}
        /> 
      </View>

      <View style={styles.reviewSection}>
        <Text style={styles.surveyText}>How good is your skin condition  </Text>
         <AirbnbRating
          starContainerStyle={{flexWrap: 'wrap', padding: 5}}
          count={10}
          reviews={surveyRange}
          onFinishRating={(e)=>{
            setRateValue({...rateValue, survey2: e});
          }}
          defaultRating={rateValue.survey2}
          size={40}
          showRating={false}
        /> 
      </View>
      <View style={styles.btnSection}>
          <View style={styles.btn}>
            <Button onPress={onClearSurvey} title="Clear"  />
          </View>
          <View style={styles.btn}>
            <Button onPress={onSubmitSurvey} title="Submit"  />
          </View>
       </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  reviewSection: {
    alignItems: 'center',
    paddingVertical: 30
  },
  btn: {
    paddingHorizontal:20
  },
  btnSection: {
    flexDirection:'row',justifyContent:'space-around'
  },
  
  title: {
    fontSize: 23,
    fontWeight: 'bold',
  },
  surveyText: {
    fontSize: 18,
  },
});

export default App;
