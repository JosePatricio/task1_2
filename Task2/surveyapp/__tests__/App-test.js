/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {useService} from '../src/hooks/useService';


jest.mock("@react-native-community/async-storage", () =>
  require("@react-native-community/async-storage/jest/async-storage-mock"),
);


