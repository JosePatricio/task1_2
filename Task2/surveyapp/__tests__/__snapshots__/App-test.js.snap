// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
      }
    }
  >
    Mobile Survey App
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingVertical": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      How was your sleep last night
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingVertical": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      How good is your skin condition  
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Clear
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
